Add support for labels to more K8sResource types

MujibAzizi opened the pull request:
This PR adds support for labels to more classes which extend the `K8sResource` class.

Right now, calling `setLabels` on, for example, `K8sService` leads to the `HasAttributes`'s `__call` method being used which adds the `labels` as a top level key rather than as a child of `metadata`. To correct the behaviour of this, the `HasLabels` must be added to all classes.

**Classes to add `HasLabels`, tests and update documentation for:**
- [x] K8sClusterRoleBinding
- [x] K8sConfigMap
- [x] K8sHorizontalPodAutoscaler
- [x] K8sIngress
- [x] K8sNamespace
- [x] K8sPersistentVolume
- [x] K8sPersistentVolumeClaim
- [x] K8sRole
- [x] K8sRoleBinding
- [x] ~K8sScale~ Scale is not an Kubernetes resource and thus I will not add `HasLabels` to it.
- [x] K8sSecret
- [x] K8sService
- [x] K8sServiceAccount
- [x] K8sStorageClass

Closes https://github.com/renoki-co/php-k8s/issues/52

rennokki commented:
Interestingly enough, I think there are more resources in this situation. 🤔

MujibAzizi commented:
I will be adding it to all resources, and if that works out we can discuss to move `HasLabels` to `K8sResource`.

If you run the tests locally, does it work for you?

MujibAzizi commented:
FYI: I am going to switch to use Minikube, Docker for Mac has different default.

For example; there is no storageclass called `standard` available. To fix this is not in scope of this PR.

rennokki commented:
@MujibAzizi I use Minikube for testing, maybe that's what the testing is biased towards it.

If all resources accept Labels, then it can be moved to the K8sResource. However, I think it's not the same as the annotations.

MujibAzizi commented:
@rennokki I've finished adding `HasLabels` to the required classes and added the tests. Also, I've added some generic documentation.

I'm marking the PR as Ready for Review.

rennokki commented:
Thank you for your contribution. ❤️

MujibAzizi commented:
> Thank you for your contribution. ❤️

My pleasure! Thank you for merging.
